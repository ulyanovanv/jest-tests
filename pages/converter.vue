<template>
  <v-layout>
    <v-flex class="text-center">
      <section v-if="rates">
        <div class="currency-converter">
          <input-box inputbox="1"/>
          <switcher></switcher>
          <input-box inputbox="2"/>
        </div>
      </section>

      <section v-else>
        <p class="header error">Unfortunately, no currency rates are available at the moment. Try again later.</p>
      </section>
    </v-flex>
  </v-layout>
</template>

<script>
import { mapGetters } from "vuex";

import InputBox from '@/components/InputBox.vue'
import Switcher from '@/components/Switcher.vue'

export default {
  name: 'Exchanger',
  components: {
    InputBox,
    Switcher
  },

  async asyncData({ store }) {
    await store.dispatch('rates/getRates')
  },

  computed: {
    ...mapGetters({
      rates: "rates/getRates"
    })

    // ratesObj() {
    //   return Object.entries(this.rates).length > 0
    // }
  }
}
</script>
