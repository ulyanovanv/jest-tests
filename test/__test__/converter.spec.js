import { mount, createLocalVue, shallowMount} from '@vue/test-utils'
import converter from '@/pages/converter.vue'
import Vuex from 'vuex'
import Vuetify from 'vuetify'
import Vue from 'vue';

const localVue = createLocalVue()
Vue.use(Vuex)
Vue.use(Vuetify)

describe('converter', () => {
  let getters, store

  beforeEach(() => {
    getters = {
      'rates/getRates': () => {
        return {
          'EUR': 1,
          'USD': 1.11
        }
      },
      'rates/getCurrency_1': () => 'EUR',
      'rates/getCurrency_2': () => 'USD',

      'rates/getValue_1': () => 1,
      'rates/getValue_2': () => 1.11
    }

    store = new Vuex.Store({
      getters
    })
  })

  test('is a Vue instance', () => {
    const wrapper = mount(converter, { store, Vue })
    expect(wrapper.isVueInstance()).toBeTruthy()
  })

  test('Snapshot: should render content correctly', () => {
    const wrapper = mount(converter, {
      store,
      Vue
    })

    expect(getters['rates/getRates']()['EUR']).toEqual(1)
    expect(wrapper.vm.$el).toMatchSnapshot()
  })

  test("Should return exchanger interface, if rates are passed", () => {
    const wrapper = mount(converter, {
      store,
      Vue
    })

    expect(wrapper.contains('.currency-converter')).toBe(true)
  })

  test("Should return error text, if rates are empty object", () => {
    store = new Vuex.Store({
      getters: {
        ...getters,
        'rates/getRates': () => {
          return {}
        }
      }
    })

    const wrapper = mount(converter, {
      store,
      Vue
    })

    console.log(getters['rates/getRates']())

    expect(wrapper.contains('.error')).toBe(true)
  })
})
