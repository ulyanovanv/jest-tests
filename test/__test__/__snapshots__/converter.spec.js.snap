// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`converter Snapshot: should render content correctly 1`] = `
<div
  class="layout"
>
  <div
    class="flex text-center"
  >
    <section>
      <div
        class="currency-converter"
      >
        <div
          class="input-box"
        >
          <span
            class="input-box_symbol"
          >
            â‚¬
          </span>
           
          <input
            id="valueNum"
            type="text"
          />
           
          <select>
            <option>
              EUR
            </option>
            <option>
              USD
            </option>
            <option>
              GBP
            </option>
          </select>
        </div>
         
        <div
          class="switcher"
          id="switcher"
        >
          <img
            class="rotate"
            src="@/assets/images/switch.svg"
          />
        </div>
         
        <div
          class="input-box"
        >
          <span
            class="input-box_symbol"
          >
            $
          </span>
           
          <input
            id="valueNum"
            type="text"
          />
           
          <select>
            <option>
              EUR
            </option>
            <option>
              USD
            </option>
            <option>
              GBP
            </option>
          </select>
        </div>
      </div>
    </section>
  </div>
</div>
`;
